frontend/src/components/dashboard/create/email/EmailRecipients.tsx

```diff
@@ -157,26 +121,56 @@ const EmailRecipients = ({
     return () => clearTimeout(timeoutId)
   }, [campaignId, csvFilename, forceReset, isCsvProcessing, isMounted])
 
+  // Select managed list
+  useEffect(() => {
+    const setSelectedList = async () => {
+      try {
+        // trigger change only if it isn't already the current one
+        const currentValue = phonebookList.filter(
+          (l) => l.label === csvInfo.csvFilename?.replace('.csv', '')
+        )[0]?.value
+        console.log(currentValue)
+        console.log(selectedPhonebookListId)
+        if (
+          selectedPhonebookListId &&
+          selectedPhonebookListId !== +currentValue
+        ) {
+          await selectPhonebookList({
+            campaignId: +(campaignId as string),
+            listId: selectedPhonebookListId,
+          })
+          setIsCsvProcessing(true)
+        }
+      } catch (e) {
+        setErrorMessage((e as Error).message)
+      }
+    }
+
+    void setSelectedList()
+  }, [campaignId, selectedPhonebookListId])
+
   // If campaign properties change, bubble up to root campaign object
   useEffect(() => {
     updateCampaign({
       isCsvProcessing,
       csvFilename,
       numRecipients,
-      shouldSaveList,
     })
-  }, [
-    isCsvProcessing,
-    csvFilename,
-    numRecipients,
-    updateCampaign,
-    shouldSaveList,
-  ])
+  }, [isCsvProcessing, csvFilename, numRecipients, updateCampaign])
 
-  // If shouldSaveList is modified, send info to backend
+  const retrieveAndPopulatePhonebookLists = useCallback(async () => {
+    const lists = await getPhonebookListsByChannel({ channel: campaign.type })
+    setPhonebookList(
+      lists.map((l: AgencyList) => {
+        // replace all space with dash
+        return { label: l.name.replaceAll(' ', '-'), value: l.id.toString() }
```

Comment:
why "replace all space with dash" ? Shouldnt the user see the same name they created the list as

Reply:
this is what phonebook has returned 
<img width="879" alt="Screenshot 2023-06-05 at 1 33 10 PM" src="https://github.com/opengovsg/postmangovsg/assets/74128885/0e8a7788-b6f6-407a-8949-771114c2e397">

Reply:
it seems that there is an inconsistency with sending a GET request to retrieve the s3 file vs what is stored in the phonebook db.

Reply:
Thats the file name. Since phonebook doesnt store the file name, it creates on the fly using list name ( not sure why postman needs it but this on the fly computation was done for postman only)

The list name is as is given by the user and postman should display the same

Reply:
Two apis for postman UI

Get /managed-list?owner= ==> To get the lists owned by the owner
Get /managed-list/:listId/members/s3  ==> Once user selects this and moves to next screen or equivalent, get this s3 url

Reply:
Postman UI displays the file that was uploaded in the section below

![Screenshot 2023-06-05 at 1 46 17 PM](https://github.com/opengovsg/postmangovsg/assets/74128885/4b83e4bb-7daf-4492-9eda-8234f5569ae1)

and the above change was intended to make things consistent with the UI

Reply:
fix applied : remove the replacement of spaces with dashes. but the bottom portion will be left untouched as that is indeed a filename